libs/libxx/libcxxmini/libxx_cxa_guard.cxx

```diff
@@ -44,7 +44,7 @@ typedef int __guard;
 // variable and the first byte contains 0 prior to initialization, and
 // 1 after.
 
-__extension__ typedef int __guard __attribute__((mode(__DI__)));
```

Comment:
why not define to empty in compiler.h?

Reply:
> why not define to empty in compiler.h?

is that mean to keep "libs/libxx/libcxxmini/libxx_cxa_guard.cxx" left unchanged, and add a following empty define in "nuttx/compiler.h":
`#  define __extension__`

Reply:
Yes, let only defined in special compilers, such as GHS